clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -628,6 +628,15 @@ static ParamDesc makeParamDesc(const FieldDecl *Src, QualType Ty) {
                          Ctx.getTrivialTypeSourceInfo(Ty));
 }
 
+// Creates a parameter descriptor for kernel object
+static ParamDesc makeParamDesc(const CXXRecordDecl *Src, QualType Ty) {
```

Comment:
Make the above a template and combine these two functions :)

Reply:
I didn't do this since the body is now different.